src/css/wizard-navigation-bar.scss

```diff
@@ -0,0 +1,394 @@
+      @content;
+    }
+    &.navigable.optional a:hover .step-indicator {
+      $wz-param-indicator-state: 'hover' !global;
+      $wz-param-indicator-color: $wz-color-optional !global;
+      @content;
+    }
+    &.navigable.done, &.navigable.completed {
+      a:hover .step-indicator {
+        $wz-param-indicator-state: 'hover' !global;
+        $wz-param-indicator-color: $wz-color-done !global;
+        @content;
+      }
+    }
+    &.navigable.current a:hover .step-indicator {
+      $wz-param-indicator-state: 'hover' !global;
+      $wz-param-indicator-color: $wz-color-current !global;
+      @content;
+    }
+    &.navigable.current.done, &.navigable.current.completed {
+      a:hover .step-indicator {
+        $wz-param-indicator-state: 'hover' !global;
+        $wz-param-indicator-color: $wz-color-editing !global;
+        @content;
+      }
+    }
+  }
+}
+
+
+// Define navigation bar layout
+//
+// Arguments:
+//   $layout - 'horizontal' | 'vertical'
+//
+@mixin wz-define-layout($layout) {
+  aw-wizard-navigation-bar.#{$layout} {
+
+    &.small ul.steps-indicator {
+      @include wz-define-style($layout, $width: $wz-small-dot-width, $height: $wz-small-dot-height, $border-width: 0) {
+        @include wz-state-circle-with-background($wz-param-indicator-color);
+      }
+    }
+
+    &.large-filled ul.steps-indicator {
+      @include wz-define-style($layout, $border-width: 0) {
+        @include wz-state-circle-with-background($wz-param-indicator-color);
+      }
+    }
+
+    &.large-empty ul.steps-indicator {
+      @include wz-define-style($layout) {
+        @include wz-state-circle-with-border($wz-param-indicator-color);
+      }
+    }
+
+    &.large-filled-symbols ul.steps-indicator {
+      @include wz-define-style($layout, $border-width: 0) {
+        @include wz-state-circle-with-background-and-content($wz-param-indicator-color);
+      }
+    }
+
+    &.large-empty-symbols ul.steps-indicator {
+      @include wz-define-style($layout) {
+        @include wz-state-circle-with-border-and-content($wz-param-indicator-color);
+      }
+    }
+  }
+}
+
+aw-wizard-navigation-bar {
+
+  ul.steps-indicator {
+
+    // http://www.paulirish.com/2012/box-sizing-border-box-ftw/
+    * {
+      -webkit-box-sizing: border-box;
+      -moz-box-sizing: border-box;
+      box-sizing: border-box;
+    }
+
+    li {
+      position: relative;
+      pointer-events: none;
+
+      a .label {
+        color: $wz-color-current;
+        line-height: $wz-text-height;
+        font-size: $wz-text-height;
+        text-decoration: none;
+        text-transform: uppercase;
+        font-weight: bold;
+        transition: 0.25s;
+      }
+    }
+
+    li.navigable {
+      pointer-events: auto;
+
+      a {
+        cursor: pointer;
+      }
+
+      a:hover .label {
+        color: darken($wz-color-current, 20%);
+      }
+    }
+  }
+}
+
+@include wz-define-layout('horizontal');
+@include wz-define-layout('vertical');
```

Comment:
What exactly does this `mixin` define? I think the mixin needs more comments to make this clear, especially where the dividing line between the purpose of the mixin and the following style blocks is located.

Reply:
I have converted this mixin to an SCSS loop (`each`) in 6509931. The new version looks more readable to me.